Avoid issue with option hash removing :filter.

mdave opened the pull request:
I'm working on a small feature for jekyll-scholar which relies on `to_s` for `Entry` being passed a `:filter` symbol through the `options` array. I noticed that the filter is only being applied to the first field in each entry, which I believe is due to the `options.delete` killing this on the first value being processed.

inukshuk commented:
Good catch, thanks!

Trying to keep the change to a minimum, would it work to change the original line to something like:

```Ruby
return convert(options[:filter]).to_s(options.reject { |k,| k == :filter }) if options.key?(:filter)
```

mdave commented:
So I just tried this and it does _nearly_ the same thing; the issue is that with just the one line change, when you then call `convert()` which dup's the value, you lose information of whether the value is a symbol and then end up with output like:

```
month = {dec},
```
instead of
```
month = dec,
```

where `dec` is the bibtex token. The second commit in the MR is an attempt to fix that in what I _think_ is the right way, but possibly not! If so I think you could modify your one liner a bit although it is a bit ugly:

```ruby
return convert(options[:filter]).to_s(options.reject { |k,| k == :filter || (k == :quotes && (!atomic? || symbol?)) }) if options.key?(:filter)
```

mapreal19 commented:
@mdave it could be useful to reproduce the bug in tests here too: https://github.com/inukshuk/bibtex-ruby/blob/master/test/bibtex/test_value.rb#L86-L103

mdave commented:
@mapreal19 Good idea -- I've added a test in `test_entry.rb` since the issue applies to when you have multiple values in the entry. I've also changed to the one-liner in my comment above.

inukshuk commented:
Great work, thanks!

mapreal19 commented:
@mdave awesome stuff!
